src/qt/overviewpage.cpp

```diff
@@ -299,6 +302,16 @@ void OverviewPage::changeEvent(QEvent* e)
     QWidget::changeEvent(e);
 }
 
+// Only show most recent NUM_ITEMS rows
+void OverviewPage::LimitTransactionRows()
+{
+    if (filter && ui->listTransactions && ui->listTransactions->model() && filter.get() == ui->listTransactions->model()) {
+        for (int i = 0; i < filter->rowCount(); ++i) {
+            ui->listTransactions->setRowHidden(i, i >= NUM_ITEMS);
+        }
```

Comment:
Could this cause any performance issues with 100K+ txs wallets?

Reply:
I loaded a wallet with 30K txs and ran a time profiler. This call took approximately 3 milliseconds:

<img width="998" alt="image" src="https://user-images.githubusercontent.com/116917595/216152217-c6c4d7cf-434c-44c7-88f8-fa8c97fbfee5.png">

The vast majority of time was consumed by re-running the filter and converting to `QDateTime` (both already existing behavior and unrelated to this PR):

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/116917595/216153223-5876a1e6-e519-4562-b4ce-48a46fa05bb9.png">

Qualitatively, I didn't notice any difference at all.

I'm happy to generate a bunch more transactions overnight and report back tomorrow.